src/GafferScene/LightToCamera.cpp

```diff
@@ -0,0 +1,312 @@
+			ConstStringDataPtr type = Metadata::value<StringData>( metadataTarget, "type" );
+
+			if( !type || !shaderParameters )
+			{
+				continue;
+			}
+
+			IECore::CameraPtr result = new IECore::Camera();
+
+
+			float screenWindowScale = 1.0f;
+			if( type->readable() == "distant" )
+			{
+			    const float locatorScale = parameter<float>( metadataTarget, shaderParameters, "locatorScaleParameter", 1 );
+				
+				result->parameters()["projection"] = new StringData( "orthographic" );
+				result->parameters()["clippingPlanes"] = new V2fData( V2f( -100000, 100000 ) );
+				screenWindowScale = locatorScale;
+			}
+			else if( type->readable() == "spot" )
+			{
+
+				/// \todo - this logic should look much cleaner once we have LightAccessor
+				float coneAngle = parameter<float>( metadataTarget, shaderParameters, "coneAngleParameter", 0.0f );
+				float penumbraAngle = parameter<float>( metadataTarget, shaderParameters, "penumbraAngleParameter", 0.0f );
+
+				if( ConstStringDataPtr angleUnit = Metadata::value<StringData>( metadataTarget, "angleUnit" ) )
+				{
+					if( angleUnit->readable() == "radians" )
+					{
+						coneAngle *= 180.0 / M_PI;
+						penumbraAngle *= 180 / M_PI;
+					}
+				}
+
+				const std::string *penumbraType = NULL;
+				ConstStringDataPtr penumbraTypeData = Metadata::value<StringData>( metadataTarget, "penumbraType" );
+				if( penumbraTypeData )
+				{
+					penumbraType = &penumbraTypeData->readable();
+				}
+
+				float outerAngle = coneAngle;
+				if( penumbraType && *penumbraType == "outset" )
+				{
+					outerAngle = coneAngle + 2.0f * penumbraAngle;
+				}
+				outerAngle = std::max( 0.25f, outerAngle );
+
+
+				if( parameter<bool>( metadataTarget, shaderParameters, "lensRadiusEnableParameter", true ) )
+				{
+					float lensRadius = parameter<float>( metadataTarget, shaderParameters, "lensRadiusParameter", 0.0f );
+					float focalPointOffset = lensRadius / tan( 0.5f * outerAngle * M_PI / 180.0f );
+
+					M44f transformOffset;
+					transformOffset.setTranslation( V3f( 0.0f, 0.0f, focalPointOffset ) );
+					result->parameters()["__projectionTransformOffset"] = new M44fData( transformOffset );
+					result->parameters()["clippingPlanes"] = new V2fData( V2f( focalPointOffset, 100000 ) );
+
+				}
+				result->parameters()["projection"] = new StringData( "perspective" );
+				result->parameters()["projection:fov"] = new FloatData( outerAngle );
+			}
+			else
+			{
+				return NULL;
+			}
+
+			V2i res2( resolution, resolution );
+			result->parameters()["resolutionOverride"] = new V2iData( res2 );
+			Box2f screenWindow( screenWindowScale * V2f( -1.0f ), screenWindowScale * V2f( 1.0f ) );
+			result->parameters()["screenWindow"] = new Box2fData( screenWindow );
+
+			return result;
+
+		}
+
+		return NULL;
+	}
+}
+
+IE_CORE_DEFINERUNTIMETYPED( LightToCamera );
+
+size_t LightToCamera::g_firstPlugIndex = 0;
+
+LightToCamera::LightToCamera( const std::string &name )
+	:	SceneElementProcessor( name )
```

Comment:
Prefer `SceneElementProcessor( name, GafferScene::Filter::NoMatch )`, to create a node which requires a filter before it does anything. We have a todo item to do this for all existing nodes, so in the meantime we shouldn't add any more with the wrong behaviour.